src/main/java/me/karakelley/http/HttpServer.java

```diff
@@ -0,0 +1,37 @@
+package me.karakelley.http;
+
+import java.io.*;
+import java.net.ServerSocket;
+
+
+class HttpServer {
+  private final int port;
+  private final ServerLogger logger;
+  private ServerSocket serverSocket;
+  private Integer portNumber;
+
+  public HttpServer(int port, ServerLogger logger) {
+    this.port = port;
+    this.logger = logger;
+  }
+
+  public void start() {
+    try (ServerSocket serverSocket = new ServerSocket(port)) {
+      setPortNumber(serverSocket);
```

Comment:
Any time you can avoid state like this is a win. It makes the code harder to reason about. The only interface your `HttpServer` needs to adhere to is `getPortNumber`. Why not just ask the server what the local port is in that getter (that's the beauty of getters if there is one).

Reply:
I was having issues using a getter using try-with-resources because the `serverSocket` was being initialized within the try-with-resources causing a null pointer exception. This was what I could come up with to make it work with try-with-resources. I just changed it to use a getter with a `finally` block instead.